internal/mode/static/metrics/collector.go

```diff
@@ -0,0 +1,121 @@
+package metrics
+
+import (
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// ManagerCollector is an interface for the metrics of the Nginx runtime manager
+type ManagerCollector interface {
```

Comment:
I think this interfaces should belong to the runtime package, where the runtime manager is defined. Because the manager really defines which metrics it collects.